### PR TITLE
prov/rxd: Added check function for rxd_remote_cq_data

### PR DESCRIPTION
**Pull request by nikitaxgusev**
Generalized checking of cq_data for all transmit operations

Signed-off-by: Nikita Gusev <nikita.gusev@intel.com>

### CODE REVIEW

**Comment by rajachan** [comment]
add to whitelist

File: prov/rxd/src/rxd.h [hunk 1]
```diff
@@ -427,6 +427,14 @@ void rxd_init_atom_hdr(void **ptr, enum fi_datatype datatype,
 		       enum fi_op atomic_op);
 size_t rxd_init_msg(void **ptr, const struct iovec *iov, size_t iov_count,
 		    size_t total_len, size_t avail_len);
+static inline size_t rxd_check_init_cq_data(void **ptr, struct rxd_x_entry *tx_entry,
+			      		    size_t max_inline)
```
Comment (shefty) [inline prov/rxd/src/rxd.h]: I'm not sure what the compiler will do.  So to be safe, pass in &max_inline, and only update it if needed.  The function can convert to void.  Best case is that this saves writing to max_inline when it isn't changing.  The function would end up being almost an exact copy of the code that it is replacing.

Comment (nikitaxgusev) [reply prov/rxd/src/rxd.h]: @shefty Thanks for review. I passed in `&max_inline`, the function was converted in `void` one. Please, check it out.

**Comment by shefty** [comment]
thanks!

**Pull request merged**